include/swift/AST/Expr.h

```diff
@@ -953,15 +953,23 @@ class TapExpr : public Expr {
 class InterpolatedStringLiteralExpr : public LiteralExpr {
   /// Points at the beginning quote.
   SourceLoc Loc;
+  /// Points at the ending quote.
+  /// Needed for the upcoming ASTScope subsystem because lookups can be targeted
+  /// to inside an InterpolatedStringLiteralExpr.
+  SourceLoc TrailingQuoteLoc;
```

Comment:
I think it might be better to just make these the `StartLoc` and `EndLoc`. (But I don't know if that might break something that assumes literals have only one location.)

Reply:
@rintaro What do you think? Will this break something?

Reply:
@brentdax Yeah, I was hoping to avoid the potential destabilization. But it's worth exploring. I'll try it and see what breaks.

Reply:
Sorry, we can't do that.
If we make this `EndLoc`, `Lexer::getLocForEndOfToken()` won't work for `stringLiteral->getEndLoc()`. Because it tries to lex the end quote as the start of of the string literal.

Reply:
Ouch.

Reply:
Thanks, @rintaro.